### PR TITLE
Implementation done for spring crud

### PR DESCRIPTION
**Pull request by cristicojo**

### CODE REVIEW

File: src/main/java/todolist/controller/ToDoController.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import todolist.service.ToDoService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
+@AllArgsConstructor
+public class ToDoController {
+
+    private ToDoService service;
```
Comment (TicoiuLucian) [inline src/main/java/todolist/controller/ToDoController.java]: private final + `@RequiredArgsConstructor`

File: src/main/java/todolist/controller/ToDoController.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+
+    @GetMapping("/all")
+    public List<ToDo> getAll() {
```
Comment (TicoiuLucian) [inline src/main/java/todolist/controller/ToDoController.java]: Use ResponseEntity in all endpoints

File: src/main/java/todolist/repository/ToDoRepository.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package todolist.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import todolist.entity.ToDo;
+
+@Repository
```
Comment (TicoiuLucian) [inline src/main/java/todolist/repository/ToDoRepository.java]: No need for `@Repository` here. JpaRepository already contains one

File: src/main/java/todolist/service/impl/ToDoServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import todolist.service.ToDoService;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class ToDoServiceImpl implements ToDoService {
+
+    private ToDoRepository repository;
```
Comment (TicoiuLucian) [inline src/main/java/todolist/service/impl/ToDoServiceImpl.java]: Same as controller, private final + `@RequiredArgsConstructor`

File: src/main/java/todolist/service/impl/ToDoServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        newToDo.setDueDate(toDoRequest.getDueDate());
+        newToDo.setStatus(toDoRequest.getStatus());
+
+        return repository.save(newToDo);
+    }
+
+
+    @Override
+    public ToDo create(ToDo toDoRequest) {
+        return repository.save(toDoRequest);
```
Comment (TicoiuLucian) [inline src/main/java/todolist/service/impl/ToDoServiceImpl.java]: What happens if save fails? Let's say a constraint on an attribute is not satisfied, like `unique`

**Review by TicoiuLucian — APPROVED** [review]
Keep up the good work

File: src/main/java/todolist/controller/ToDoController.java (hunk 1 above)
Comment (cristicojo) [reply src/main/java/todolist/controller/ToDoController.java]: done

File: src/main/java/todolist/controller/ToDoController.java (hunk 2 above)
Comment (cristicojo) [reply src/main/java/todolist/controller/ToDoController.java]: done

File: src/main/java/todolist/service/impl/ToDoServiceImpl.java (hunk 4 above)
Comment (cristicojo) [reply src/main/java/todolist/service/impl/ToDoServiceImpl.java]: done

File: src/main/java/todolist/service/impl/ToDoServiceImpl.java (hunk 5 above)
Comment (cristicojo) [reply src/main/java/todolist/service/impl/ToDoServiceImpl.java]: The ToDoExceptionHandler in the exceptionhandler package will take care of all validations and constraints before calling save() method

**Pull request merged**